device/core/src/blob_upload/file_upload_api.ts

```diff
@@ -37,107 +37,110 @@ export class FileUploadApi implements FileUploadInterface {
-
-        this._authenticationProvider.getDeviceCredentials((err, deviceCredentials) => {
-            /*Codes_SRS_NODE_FILE_UPLOAD_ENDPOINT_16_013: [`notifyUploadComplete` shall create a `POST` HTTP request to a path formatted as the following:`/devices/URI_ENCODED(<deviceId>)/files/<correlationId>?api-version=<api-version>`]*/
-            const path = endpoint.devicePath(encodeUriComponentStrict(deviceCredentials.deviceId)) + '/files/notifications/' + encodeURIComponent(correlationId) + endpoint.versionQueryString();
-            const body = JSON.stringify(uploadResult);
-
-            /*Codes_SRS_NODE_FILE_UPLOAD_ENDPOINT_16_014: [The `POST` HTTP request shall have the following headers:
-            ```
-            Host: <hostname>,
-            Authorization: <sharedAccessSignature>,
-            'User-Agent': <version>,
-            'Content-Type': 'application/json; charset=utf-8',
-            'Content-Length': <content length>,
-            'iothub-name': <hub name>
-            ```]*/
-            let headers: any = {
-                Host: deviceCredentials.host,
-                'User-Agent': packageJson.name + '/' + packageJson.version,
+    notifyUploadComplete(correlationId: string, uploadResult: BlobUploadResult, done?: ErrorCallback): Promise<void> | void {
+        return errorCallbackToPromise((_callback) => {
+            /*Codes_SRS_NODE_FILE_UPLOAD_ENDPOINT_16_010: [`notifyUploadComplete` shall throw a `ReferenceError` if `correlationId` is falsy.]*/
+            /*Codes_SRS_NODE_FILE_UPLOAD_ENDPOINT_16_012: [`notifyUploadComplete` shall throw a `ReferenceError` if `uploadResult` is falsy.]*/
+            if (!correlationId) throw new ReferenceError('correlationId cannot be \'' + correlationId + '\'');
+            if (!uploadResult) throw new ReferenceError('uploadResult cannot be \'' + uploadResult + '\'');
+
+            this._authenticationProvider.getDeviceCredentials((err, deviceCredentials) => {
+                /*Codes_SRS_NODE_FILE_UPLOAD_ENDPOINT_16_013: [`notifyUploadComplete` shall create a `POST` HTTP request to a path formatted as the following:`/devices/URI_ENCODED(<deviceId>)/files/<correlationId>?api-version=<api-version>`]*/
+                const path = endpoint.devicePath(encodeUriComponentStrict(deviceCredentials.deviceId)) + '/files/notifications/' + encodeURIComponent(correlationId) + endpoint.versionQueryString();
+                const body = JSON.stringify(uploadResult);
+
+                /*Codes_SRS_NODE_FILE_UPLOAD_ENDPOINT_16_014: [The `POST` HTTP request shall have the following headers:
+                ```
+                Host: <hostname>,
+                Authorization: <sharedAccessSignature>,
+                'User-Agent': <version>,
                 'Content-Type': 'application/json; charset=utf-8',
-                'Content-Length': body.length,
-                'iothub-name': deviceCredentials.host.split('.')[0]
```

Comment:
I'm concerned that these comments duplicate the code that is below. That seems like it will be confusing later on when the code changes but the comment doesn't.

Reply:
Won't fix.

---
In reply to: [218886662](https://github.com/Azure/azure-iot-sdk-node/pull/372#discussion_r218886662) [](ancestors = 218886662)